### PR TITLE
Use layouts in legacy pre-processing

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Use layouts for legacy preprocessing in MO*

### Tickets:
 - *73719*

### CODE REVIEW

**Review by evolosen — APPROVED** [review]

**Review by popovaan — APPROVED** [review]

File: tools/mo/openvino/tools/mo/middle/passes/infer.py [hunk 1]
```diff
@@ -220,10 +220,33 @@ def override_batch(graph: Graph, batch: int):
...
+            if not node.soft_get('fixed_batch', False):
+                name = node.soft_get('name', node.id)
+                if layout_values and name in layout_values and layout_values[name]['source_layout']:
+                    from openvino.runtime import Layout  # pylint: disable=no-name-in-module,import-error
+
+                    layout = layout_values[name]['source_layout']
+                    layout_parsed = Layout(layout)
+                    if layout_parsed.has_name('N'):
+                        idx = layout_parsed.get_index_by_name('N')
+                        node['shape'][idx] = batch
```
Comment (sadolini) [inline tools/mo/openvino/tools/mo/middle/passes/infer.py:242]: Why we make channel index positive (convert negative to positive) and batch index we don't convert?

File: tools/mo/openvino/tools/mo/middle/passes/infer.py [hunk 2]
```diff
@@ -220,10 +220,33 @@ def override_batch(graph: Graph, batch: int):
...
     """
     if batch is not None:
-        for node_id, data in graph.nodes(data=True):
-            if 'op' in data and data['op'] == 'Parameter' and not data.get('fixed_batch', False):
-                validate_batch_in_shape(data['shape'], data['name'])
-                data['shape'][0] = batch
+        in_nodes = graph.get_op_nodes(op='Parameter')
+        layout_values = None
+        if 'layout_values' in graph.graph['cmd_params']:
+            layout_values = graph.graph['cmd_params'].layout_values.copy()
```
Comment (sadolini) [inline tools/mo/openvino/tools/mo/middle/passes/infer.py:226]: minor: the same code in 2 places, for me looks better if move it to separate function

**Review by sadolini — APPROVED** [review]
LGTM Just couple minor questions.

File: tools/mo/openvino/tools/mo/middle/passes/infer.py (hunk 1 above)
Comment (mvafin) [reply tools/mo/openvino/tools/mo/middle/passes/infer.py:242]: Here it is obvious that negative index will work as len(shape)-idx, for channels it was done in special function so I decided to normalize index

**Comment by ilya-lavrenov** [comment]
> thirdparty/open_model_zoo

looks like not needed change

**Review by rkazants — APPROVED** [review]

**Pull request merged**